### PR TITLE
Update Scene Assets search

### PR DESCRIPTION
**Pull request by CITIZENDOT**
## Summary

- Currently, Scene Asset search will work only if a category (Model, Image, Material etc..) is selected. This PR broadens the search to all assets, when no category is selected.
- Also adds `WAV` and `JSON` AssetTypes.

## References
closes #_insert number here_

### CODE REVIEW

File: packages/editor/src/components/assets/SceneAssetsPanel.tsx [hunk 1]
```diff
@@ -251,20 +251,26 @@ const SceneAssetsPanel = () => {
...
-        return (
-          <>
-            {searchedStaticResources
-              .filter((resource) => {
+    if (searchedStaticResources.value) {
+      return (
+        <>
+          {searchedStaticResources
+            .filter((resource) => {
+              if (selectedCategory.value !== null) {
```
Comment (aditya-mitra) [inline packages/editor/src/components/assets/SceneAssetsPanel.tsx]: This case will be true for empty string as well.
We can do this:  `if (selectedCategory.value?.name && selectCategory.value?.object)`

File: packages/editor/src/components/assets/SceneAssetsPanel.tsx [hunk 2]
```diff
@@ -251,20 +251,26 @@ const SceneAssetsPanel = () => {
...
-        )
+              } else if (searchText.value === '') {
+                return false
+              } else {
+                return true
+              }
+            })
+            .map((resource) => {
+              return <ResourceFile key={resource.value.id} resource={resource.get(NO_PROXY) as StaticResourceType} />
+            })}
```
Comment (aditya-mitra) [inline packages/editor/src/components/assets/SceneAssetsPanel.tsx]: we can make this shorter
```tsx
.map((resource) => (<ResourceFile key={resource.value.id} resource={resource.get(NO_PROXY) as StaticResourceType} />
            ))}

**Review by aditya-mitra — CHANGES_REQUESTED** [review]
Looks good! Just a few changes and let's merge it.

File: packages/editor/src/components/assets/SceneAssetsPanel.tsx (hunk 1 above)
Comment (CITIZENDOT) [reply packages/editor/src/components/assets/SceneAssetsPanel.tsx]: Done!

File: packages/editor/src/components/assets/SceneAssetsPanel.tsx (hunk 2 above)
Comment (CITIZENDOT) [reply packages/editor/src/components/assets/SceneAssetsPanel.tsx]: Done!

**Review by aditya-mitra — APPROVED** [review]
Great!

**Pull request merged**